### PR TITLE
[REF] 8.0 mgmtsystem hazard

### PR DESCRIPTION
**Pull request by llacroix**
Changed nothing and it seems to work

Depends on:
- [x] #20

### CODE REVIEW

**Comment by bwrsandman** [comment]
Needs tests

**Comment by llacroix** [comment]
Ok going to add tests

**Comment by llacroix** [comment]
The problem in this PR is that it tries to set a field as required on res_company without setting a default value during an alter table. Existing records will trigger a constraint error.

```
2015-04-01 10:12:24,162 16655 INFO 2925915-28-1a043d-all openerp.modules.module: module mgmtsystem_hazard: creating or updating database tables
2015-04-01 10:12:29,983 16655 ERROR 2925915-28-1a043d-all openerp.sql_db: bad query: ALTER TABLE "res_company" ALTER COLUMN "risk_computation_id" SET NOT NULL
Traceback (most recent call last):
      File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2925915-28-1a043d/openerp/sql_db.py", line 234, in execute
res = self._obj.execute(query, params)
IntegrityError: column "risk_computation_id" contains null values

2015-04-01 10:12:29,985 16655 WARNING 2925915-28-1a043d-all openerp.models: WARNING: unable to set column risk_computation_id of table res_company not null !
Try to re-run: openerp-server --update=module
If it doesn't work, update records and execute manually:
ALTER TABLE res_company ALTER COLUMN risk_computation_id SET NOT NULL
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2925915-28-1a043d/openerp/models.py", line 2643, in _auto_init
cr.execute('ALTER TABLE "%s" ALTER COLUMN "%s" SET NOT NULL' % (self._table, k))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2925915-28-1a043d/openerp/sql_db.py", line 158, in wrapper
return f(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/2925915-28-1a043d/openerp/sql_db.py", line 234, in execute
    res = self._obj.execute(query, params)
IntegrityError: column "risk_computation_id" contains null values
```

We have to decide between adding a default value and keeping it required=True, or set it required=False and update some views with require on certain states.

**Comment by max3903** [comment]
You can set A \* B \* C as the default value and add this configuration step in the README.rst

**Comment by bwrsandman** [comment]
Needs test for:
- mgmtsystem_hazard_risk/models/res_company.py risk_computation_id
- mgmtsystem_hazard_risk/models/mgmtsystem_hazard.py risk
- mgmtsystem_hazard_risk/models/mgmtsystem_hazard_residual_risk.py risk

**Comment by ghost** [comment]
@max3903 @bwrsandman It seems that these modules are not multi-system at all and the multi-company seems incomplete. I think that most models should have both a system_id field and a company_id field (related to the company of the system). For example, the risk formula is taken from the user's company. It would make more sence to take it from the system related to the mgmtsystem.hazard record.

**Comment by max3903** [comment]
@dufresnedavid Very true, but this is the lowest priority.

**Comment by max3903** [comment]
:+1:

File: mgmtsystem_hazard/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#
+##############################################################################
+{
+    "name": "Management System - Hazard",
+    "version": "1.1",
+    "author": "Savoir-faire Linux,Odoo Community Association (OCA)",
+    "website": "http://www.savoirfairelinux.com",
+    "license": "AGPL-3",
+    "category": "Management System",
+    "description": """\
```
Comment (yvaucher) [inline mgmtsystem_hazard/__openerp__.py]: Description must go in README.rst

File: mgmtsystem_hazard/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Management System - Hazard",
+    "version": "1.1",
```
Comment (yvaucher) [inline mgmtsystem_hazard/__openerp__.py]: Should be `8.0.1.1`

File: mgmtsystem_hazard/models/mgmtsystem_hazard.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+from openerp import models, fields
+
+
+class MgmtsystemHazard(models.Model):
+    """Hazards of the health and safety management system"""""
+
+    _name = "mgmtsystem.hazard"
+    _description = __doc__
+
+    name = fields.Char('Name', size=50, required=True, translate=True)
```
Comment (yvaucher) [inline mgmtsystem_hazard/models/mgmtsystem_hazard.py]: Size shouldn't be specified

File: mgmtsystem_hazard/models/mgmtsystem_hazard_control_measure.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        'Responsible',
+        required=True,
+    )
+    comments = fields.Text('Comments')
+    hazard_id = fields.Many2one(
+        'mgmtsystem.hazard',
+        'Hazard',
+        ondelete='cascade',
+        required=False,
+        select=True,
```
Comment (yvaucher) [inline mgmtsystem_hazard/models/mgmtsystem_hazard_control_measure.py]: In 8.0 it is now `index=True`

File: mgmtsystem_hazard/models/mgmtsystem_hazard_hazard.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+from openerp import models, fields
+
+
+class MgmtsystemHazardHazard(models.Model):
+
+    _name = "mgmtsystem.hazard.hazard"
+    _description = "Hazard"
+
+    name = fields.Char('Hazard', size=50, required=True, translate=True)
+    description = fields.Text('Description')
```
Comment (yvaucher) [inline mgmtsystem_hazard/models/mgmtsystem_hazard_hazard.py]: Tip: In such case string can be omitted as it will take field name.

Comment (yvaucher) [inline mgmtsystem_hazard/models/mgmtsystem_hazard_hazard.py]: Shouldn't description be translatable as well?

File: mgmtsystem_hazard/models/mgmtsystem_hazard_probability.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+from openerp import models, fields
+
+
+class MgmtsystemHazardProbability(models.Model):
+
+    _name = "mgmtsystem.hazard.probability"
+    _description = "Probability of hazard"
+
+    name = fields.Char('Probability', size=50, required=True, translate=True)
```
Comment (yvaucher) [inline mgmtsystem_hazard/models/mgmtsystem_hazard_probability.py]: Why not a float? or should it be a selection? A help message could be helpful

File: mgmtsystem_hazard_risk/__openerp__.py [hunk 7]
```diff
@@ -19,23 +19,30 @@
 #
 ##############################################################################
 {
-    "name": "Management System - Hazard",
+    "name": "Management System - Hazard-Risk",
     "version": "1.1",
```
Comment (yvaucher) [inline mgmtsystem_hazard_risk/__openerp__.py]: version number to `8.0.1.1`

File: mgmtsystem_hazard_risk/__openerp__.py [hunk 8]
```diff
@@ -19,23 +19,30 @@
...
     "author": "Savoir-faire Linux,Odoo Community Association (OCA)",
     "website": "http://www.savoirfairelinux.com",
     "license": "AGPL-3",
     "category": "Management System",
     "description": """\
```
Comment (yvaucher) [inline mgmtsystem_hazard_risk/__openerp__.py]: Another README.rst to make

**Comment by yvaucher** [comment]
just few remarks otherwise :+1: (code review + french translations)

**Comment by dreispt** [comment]
Multi-system and multi-company support should be mentioned in the Roadmap section, then.

File: mgmtsystem_hazard/models/mgmtsystem_hazard_control_measure.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class MgmtsystemHazardControlMeasure(models.Model):
+
+    _name = "mgmtsystem.hazard.control_measure"
+    _description = "Control Measure of hazard"
+
+    name = fields.Char(
+        'Control Measure',
+        size=50,
```
Comment (dreispt) [inline mgmtsystem_hazard/models/mgmtsystem_hazard_control_measure.py]: drop size

**Comment by dreispt** [comment]
Build green, two approvals, remarks fixed - I would merge but it needs a rebase!

**Pull request merged**